### DISCUSSION TITLE
How Can I Automatically Deploy My Website on Hostinger When Committing Changes?

### DISCUSSION THREAD

**Discussion by abdulrafaie55** [community/community#108221] [Other Feature Feedback, Questions, & Ideas] [answered]
Seeking advice on streamlining my website deployment process.
I want to connect my Hostinger-hosted website with GitHub so that every time I commit changes, the updates are deployed instantly. Can anyone guide me through the setup process for this seamless integration? Your insights and expertise are highly appreciated!

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @abdulrafaie55 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by Ariffazeel99** [comment, marked as answer]
I had this similar issue 2 weeks ago maybe that helps you,
You can visit the Hostinger documentation about the direct connection.
LINK: https://support.hostinger.com/en/articles/1583302-how-to-deploy-a-git-repository

you can follow the instructions mentioned on the page, and there is a video to help you get through easily.

in summary, there are two ways to connect either 
-public or private repository, and then you can add a webhook and enjoy continuous deployment there.
and also make sure to edit the branch name to its master by default.

I hope I answered your question. 😊

**Reply by qumberrizvi** [reply]
Does this apply to Hostinger VPS or just the shared servers? I'm unfamiliar with Hostinger and want to make sure that it's possible to push builds using webhooks for VPS as well before purchasing a plan. If so, I would be able to push builds using GitHub Actions. I need VPS specifically because the app I'm trying to deploy has some binary dependencies, and it also needs a node process manager, such as pm2. It would be very helpful if you could clarify this point. TIA.

**Reply by nandha46** [reply]
When it comes to VPS irrespective of provides, you can follow the steps provided in this guide from deployhq.
https://www.deployhq.com/blog/setting-up-git-based-deployment-on-a-virtual-private-server-vps

Hope this helps.

**Comment by queenofcorgis** [comment]
We’re glad you could get this solved and we want to make sure others can do the same so we’ve reopened this discussion for better discoverability for our members.

**Comment by facundofarias** [comment]
Probably the previous answer is the correct one, but you can also use [DeployHQ](https://www.deployhq.com/) as explained here: https://www.deployhq.com/guides/hostinger. DeployHQ allows you not only to deploy, but also to send notifications or even run commands before and after your deployments, for example, doing backups, running migrations and so on.
